### PR TITLE
feat: [CDS-76792]: create endpoint for terragrunt supported cli commands

### PR DESCRIPTION
**Pull request by vlica-harness**
## Describe your changes

The purpose of this endpoint is to get a list of terragrunt commands which are valid based on the terragrunt step as input.
For example: 
- If step is Plan, then we should get: init, workspace, plan.
- If step is Apply, then we should get: init, workspace, plan, apply.
- If step is Destroy, then we should get: init, workspace, apply, destroy.
- If step is Rollback, then we should get: init, workspace, apply, destroy.

Similar approach is done for helm and terraform.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### CODE REVIEW

**Comment by vlica-harness** [comment]
trigger uts_JDK11

**Review by tmacari-harness — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+@ApiResponses(value =
+    {
+      @ApiResponse(code = 400, response = FailureDTO.class, message = "Bad Request")
+      , @ApiResponse(code = 500, response = ErrorDTO.class, message = "Internal server error")
+    })
+@AllArgsConstructor(access = AccessLevel.PACKAGE, onConstructor = @__({ @Inject }))
+@Slf4j
+public class TerragruntResource {
+  @GET
+  @Path("terragruntCmdFlags")
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java:50]: should this API be Hidden from open api spec ?

File: 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  @ApiOperation(value = "Get Command flags based on terragrunt Step Type", nickname = "terragruntCmdFlags")
+  public ResponseDTO<Set<TerragruntCommandFlagType>> getTerragruntCommandFlags(
+      @QueryParam("stepType") @NotEmpty @NotNull String stepType) {
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java:53]: @NotNull looks redundant here, due to @NotEmpty

Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java:53]: can we create/use an enum for stepType ?

File: 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java (hunk 1 above)
Comment (vlica-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java:50]: not seeing atm a big reason to hide this.

File: 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java (hunk 2 above)
Comment (vlica-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/core/terragrunt/resources/TerragruntResource.java:53]: yeah will remove this @NotNull

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by yogesh-chauhan** [comment]
trigger review

**Pull request merged**